src/control/BBTrajectories/BBTrajectory1D.tpp

```diff
@@ -0,0 +1,210 @@
+//
+// Created by rolf on 15-01-20.
+//
+
+#include "control/BBTrajectories/BBTrajectory1D.h"
+#include <cmath>
+#include <iostream>
+#include <chrono>
+template<class num>
+num BBTrajectory1D<num>::fullBrakePos(num pos, num vel, num accMax) noexcept {
+    num acc = vel <= 0 ? accMax : - accMax;
+    num t = - vel/acc; // (inverted) time needed to break to zero velocity
+    return pos + 0.5*vel*t; // position after breaking
+}
+template<class num>
+num BBTrajectory1D<num>::accelerateBrakePos(num pos0, num vel0, num vel1, num accMax) noexcept {
+    num acc1;
+    num acc2;
+    if (vel1 >= vel0) {
+        acc1 = accMax;
+        acc2 = - accMax;
+    }
+    else {
+        acc1 = - accMax;
+        acc2 = accMax;
+    }
+    num t1 = (vel1 - vel0)/acc1; // time to reach vel1
+    num pos1 = pos0 + (0.5*(vel0 + vel1)*t1); //position at which we reach vel1
```

Comment:
Is this correct? In my mind it should be
`t1*vel0 + t1*(vel0-vel1)*0.5`

Reply:
Huh? You atleast need the pos0 somewhere so I don't see how this should work. You are driving on average (0.5*vel0+0.5*vel1) on the segment for t1 seconds so this makes a lot of sense.

Reply:
Yes,I meant pos0 + `t1*vel0 + t1*(vel1-vel0)*0.5`. However, I worked it out on paper and `t1*vel0 + t1*(vel1-vel0)*0.5` and `(0.5*(vel0 + vel1)*t1)` are equivalent